autonomy/analyse/abci/app_spec.py

```diff
@@ -205,6 +207,32 @@ def generate(self) -> Dict[str, Any]:
                 dfa_export[k] = v
         return dfa_export
 
+    @classmethod
+    def _get_reachable_states(
+        cls,
+        start_states: Set[str],
+        transition_func: Dict[Tuple[str, str], str],
+    ) -> Set[str]:
+        """Performs a BFS-like search returning the reachable set of states departing from the set of start states."""
+
+        graph: Dict[str, Set[str]] = defaultdict(set)
+        for (s1, _), s2 in transition_func.items():
+            graph[s1].add(s2)
+
+        visited: Set[str] = start_states.copy()
+        queue: List[str] = []
+        queue.extend(list(start_states))
+
+        while queue:
+            s = queue.pop(0)
```

Comment:
nice nice. Popping first element of a list is no good in python though. Use an actual queue: 
```
from collections import deque
queue = deque(list(start_states))
while queue:
    node = queue.popleft()
```

or, just call it `stack` and `stack.pop()` the last element as the ordering here is irrelevant.

Reply:
Thanks for the advice. Will fix in an upcoming PR.